src/node/blockstorage.h

```diff
@@ -48,6 +48,9 @@ extern uint64_t nPruneTarget;
 //! Check whether the block associated with this index entry is pruned or not.
 bool IsBlockPruned(const CBlockIndex* pblockindex);
 
+//! Find the last block that is pruned
+const CBlockIndex* GetLastPrunedBlock(const CBlockIndex* start_block);
```

Comment:
In commit "refactor: Introduce GetLastPruneBlock helper function" (2c815adc881fa90f14887e2ec33ccb11ef964387)

This doesn't seem to actually return the last block that is pruned. It seems to return the next block that has data. So maybe should be called `GetFirstNonPrunedBlock`? It also seems like `pruneblockchain` documentation describes this incorrectly while `getblockchaininfo` describes it correctly.

Reply:
True, I fixed this and went with with the name `GetFirstStoredBlock` inspired by the docs in `getblockchaininfo`. I also added an extra commit to fix the docs in `pruneblockchain`. I think fixing the docs is the right approach rather than change the behavior but I kept it in the separate commit in case there are different opinions on that.